### PR TITLE
private chain setup

### PR DESCRIPTION
**Pull request by jennwiederholen**
The project involves developing a `genesis_generator` that creates genesis.json files for private chains. It builds upon and enhances the Genesis file generation functionality previously available in Ethereum's `puppeth`.

After building, the tool can be executed using the command:

``` shell
$ genesis_generator 
```
or
``` shell
cmd/genesis_generator$ go run .
```

The process starts with selecting a consensus engine. Based on the chosen engine, users will be prompted to input the required options and values for that engine. The tool then allows users to either export the generated genesis.json and config.toml (optional) files or simply print them for review.

Ex)
``` shell
Which consensus engine to use? (default = Wemix)
 1. Ethash - PoW
 2. Beacon Ethash - beacon engine switched from ethash
 3. Clique - PoA
 4. Beacon Clique - beacon engine switched from clique
 5. Wbft - wemix DPoS
 6. Beacon Wbft - beacon engine switched from wbft
 7. Wemix - wemix engine swtiched from wpoa to wbft
> 7

Which accounts are allowed to seal? (mandatory at least one)
> 0xxa62987A40E094CbE020313f19F71aeeB3E48B86f
> 0xE1bD8108149FCa703B9FCD8ea967B6b7e660f13b
> 0xd19f9374f4549B2fB182ED766d6b7501494a3634
> 0x02cF1E577C79EF0E93947cCd82a4D41E0485Be73
> 0x

Want to generate config.toml file to configure static nodes to connect?
Else you have to manage peer node manually (default true)
> no

Enter timestamp you want to enable Montblanc Fork (default 1)
> 1

...
```

### CODE REVIEW

**Review by jed-wemade — APPROVED** [review]

**Comment by jennwiederholen** [comment]
[0dd1b97](https://github.com/wemixarchive/go-wemix-qbft/pull/53/commits/0dd1b97f925a61ee603ed33feaee88814b63b2d0)
I added creating address file when creating nodekey because current wbft needs nodekey's address when running the node. And I couldn't find  any geth or bootnode command to get address from private key or nodekey so,, i just added it

BUT if you think its usage is not genereal enough ( which i partially agree to 😅 ) give me your opinion please!😀

**Comment by egonspace** [comment]
Questions about the config always come up twice.

```
Want to generate config.toml file to configure static nodes to connect?
Else you have to manage peer node manually (default true)
> true
> 

Want to generate config.toml file to configure static nodes to connect?
Else you have to manage peer node manually (default true)
> 
```

or
```
Want to generate config.toml file to configure static nodes to connect?
Else you have to manage peer node manually (default true)
> 

Want to generate config.toml file to configure static nodes to connect?
Else you have to manage peer node manually (default true)
> 
```

and, genesis question too.

```
 Do you want to export generated genesis file?
 1. yes
 2. nah, just print it
> 1

Which folder to save the config.toml into? (default = current)
> 

Enter enode URLs for static nodes (press enter with empty input when done):
> 

Which accounts should be pre-funded? (advisable at least one)
> 0xaA5FAA65e9cC0F74a85b6fDfb5f6991f5C094697
> 0x

Specify your chain/network ID if you want an explicit one (default = random)
> 1115

 Do you want to export generated genesis file?
 1. yes
 2. nah, just print it

```

File: README.md [hunk 1]
```diff
@@ -239,22 +237,31 @@ aware of and agree upon. This consists of a small JSON file (e.g. call it `genes
...
+This will help you generate genesis file by simply choosing the options it gives like below : 
+``` shell
+Which consensus engine to use? (default = Wemix)
+ 1. Ethash - PoW
+ 2. Beacon Ethash - beacon engine switched from ethash
+ 3. Clique - PoA
+ 4. Beacon Clique - beacon engine switched from clique
+ 5. Wbft - wemix DPoS
+ 6. Beacon Wbft - beacon engine switched from wbft
+ 7. Wemix - wemix engine swtiched from wpoa to wbft
```
Comment (egonspace) [inline README.md]:
```suggestion
Which consensus engine to use? (default = Wbft)
 1. Ethash (proof-of-work)
 2. Beacon (proof-of-stake), merging/merged from Ethash (proof-of-work)
 3. Clique (proof-of-authority)
 4. Beacon (proof-of-stake), merging/merged from Clique (proof-of-authority)
 5. WBFT (wemix-byzantine-fault-tolerance)
 6. WBFT (wemix-byzantine-fault-tolerance), merged from Wemix3.0 (proof-of-authority)
```

File: cmd/genesis_generator/genesis_generator.go [hunk 2]
```diff
@@ -0,0 +1,390 @@
...
+		}
+		log.Info("Saved native genesis chain spec", "path", gethJson)
+	} else {
+		fmt.Println(string(out))
+	}
+}
+
+// genConfigFile creates config.toml file that defines Node.P2P.StaticNodes
+func genConfigFile() {
+	// Create a buffer to write TOML content
```
Comment (jennwiederholen) [inline cmd/genesis_generator/genesis_generator.go:311]: @egonspace 
https://github.com/wemixarchive/go-wemix-qbft/pull/53#issuecomment-2556048839
the comment you left was my mistake😅. It is fixed in here

File: README.md (hunk 1 above)
Comment (jennwiederholen) [reply README.md]: fixed in [7eab91a](https://github.com/wemixarchive/go-wemix-qbft/pull/53/commits/7eab91a423fa9b46c9de98fc5ac1df674ddada43)

**Review by egonspace — APPROVED** [review]

**Comment by egonspace** [comment]
`validators` in qbft config should be set to addresses which are same with the validators of extra field.
```
"qbft": {
      "epochLength": 30000,
      "blockPeriodSeconds": 2,
      "requestTimeoutSeconds": 4,
      "proposerPolicy": 0,
      "blockReward": "0xde0b6b3a7640000",
      "validators": null,
      "maxRequestTimeoutSeconds": null
    }
```

**Pull request merged**